Here's the full error:
```
if /bin/sh '/gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/build-aux/missing' makeinfo   -I /gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/doc \
 -o /gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/doc/m4.info /gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/doc/m4.texi; \
then \
  rc=0; \
  CDPATH="${ZSH_VERSION+.}:" && cd /gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/doc; \
else \
  rc=$?; \
  CDPATH="${ZSH_VERSION+.}:" && cd /gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/doc && \
  $restore $backupdir/* `echo ".//gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/doc/m4.info" | sed 's|[^/]*$||'`; \
fi; \
rm -rf $backupdir; exit $rc
/gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/build-aux/missing: line 81: makeinfo: command not found
WARNING: 'makeinfo' is missing on your system.
         You should only need it if you modified a '.texi' file, or
         any other file indirectly affecting the aspect of the manual.
         You might want to install the Texinfo package:
         <https://www.gnu.org/software/texinfo/>
         The spurious makeinfo call might also be the consequence of
         using a buggy 'make' (AIX, DU, IRIX), in which case you might
         want to install GNU make:
         <https://www.gnu.org/software/make/>
make[2]: *** [Makefile:1979: /gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/doc/m4.info] Error 127
make[2]: Leaving directory '/gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/spack-build/doc'
make[1]: *** [Makefile:2018: all-recursive] Error 1
make[1]: Leaving directory '/gpfs/alpine/project/proj-shared/user/spack/stage/spack-stage-m4-1.4.19-65qjzp2gyiunr7mmf233e3giax6phee4/spack-src/spack-build'
make: *** [Makefile:1974: all] Error 2
```

--- a/var/spack/repos/builtin/packages/m4/package.py
+++ b/var/spack/repos/builtin/packages/m4/package.py
@@ -3,6 +3,7 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import os
 import re
 
 
@@ -54,8 +55,8 @@ class M4(AutotoolsPackage, GNUMirrorPackage):
     @when('@1.4.19')
     def patch(self):
         """ skip texinfo of m4.info for patched m4.texi (patched only a test in it) """
-        touch('doc/m4.info')
-        touch('doc/stamp-vti')
+        timestamp = os.path.getmtime('doc/m4.info')
+        os.utime('doc/m4.texi', (timestamp, timestamp))
 
     @classmethod
     def determine_version(cls, exe):

